pkg/auth/login.go

```diff
@@ -13,9 +13,10 @@ import (
 	"k8s.io/client-go/tools/clientcmd"
 )
 
-// Login takes of authentication part and returns error if there any
-func Login(server, username, password, token, caAuth string, skipTLS bool) error {
+type KubernetesClient struct{}
```

Comment:
Can you define a factory function to return this object? [Example](https://github.com/redhat-developer/odo/blob/main/pkg/application/kubernetes.go#L20).